File: sdk/SDK.Examples/src/DocumentWorkflowExample.cs
using System;
using System.IO;
using Silanis.ESL.SDK;
using Silanis.ESL.SDK.Builder;

namespace SDK.Examples
{
	public class DocumentWorkflowExample : SDKSample
	{
        public static void Main (string[] args)
        {
            new DocumentWorkflowExample().Run();
        }

        override public void Execute() {
            DocumentPackage package = PackageBuilder.NewPackageNamed (PackageName)
					.DescribedAs ("This is a document workflow example")
					.WithSigner(SignerBuilder.NewSignerWithEmail(email1)
					            .WithFirstName("John")
					            .WithLastName("Smith"))
					.WithDocument(DocumentBuilder.NewDocumentNamed("Second Document")
                                  .FromStream(fileStream1, DocumentType.PDF)
					              .AtIndex(2)
					              .WithSignature(SignatureBuilder.SignatureFor(email1)
					              		.OnPage(0)
					               		.AtPosition(100, 100)))
					.WithDocument (DocumentBuilder.NewDocumentNamed("First Document")
                                   .FromStream(fileStream2, DocumentType.PDF)
					               .AtIndex (1)
					               .WithSignature(SignatureBuilder.SignatureFor(email1)
					               		.OnPage (0)
					               		.AtPosition (100, 100)))
					.Build ();

			packageId = eslClient.CreatePackage (package);

			Console.WriteLine("Package create, id = " + packageId);

			DocumentPackage savedPackage = eslClient.GetPackage(packageId);

			savedPackage.GetDocument("First Document").Index = 2;
            savedPackage.GetDocument("Second Document").Index = 1;

			eslClient.PackageService.OrderDocuments(savedPackage);

			Console.WriteLine("Document order saved");
		}
	}
}

File: sdk/SDK.VBExamples/SDK.VBExamples/DocumentWorkflowExample.vb
Imports System
Imports System.IO
Imports Silanis.ESL.SDK
Imports Silanis.ESL.SDK.Builder

Namespace SDK.VBExamples
    Public Module DocumentWorkflowExample
        Dim apiToken = "YOUR TOKEN HERE"
        Dim baseUrl = "ENVIRONMENT URL HERE"

        Sub Main()
            ' Create new esl client with api token and base url
            Dim client = New EslClient(apiToken, baseUrl)
            Dim file = New FileInfo(Directory.GetCurrentDirectory() + My.Resources.document)

            Dim package = PackageBuilder.NewPackageNamed("Document Workflow " + DateTime.Now) _
                .DescribedAs("This is a document workflow example") _
                .WithSigner(SignerBuilder.NewSignerWithEmail("john.smith@email.com") _
                    .WithFirstName("John") _
                    .WithLastName("Smith")) _
                .WithDocument(DocumentBuilder.NewDocumentNamed("Second Document") _
                    .FromFile(file.FullName) _
                    .AtIndex(2) _
                    .WithSignature(SignatureBuilder.SignatureFor("john.smith@email.com") _
                        .OnPage(0) _
                        .AtPosition(500, 100)) _
                    .WithSignature(SignatureBuilder.InitialsFor("john.smith@email.com") _
                        .OnPage(0) _
                        .AtPosition(500, 200)) _
                    .WithSignature(SignatureBuilder.CaptureFor("john.smith@email.com") _
                        .OnPage(0) _
                        .AtPosition(500, 300))) _
                .WithDocument(DocumentBuilder.NewDocumentNamed("First Document") _
                    .FromFile(file.FullName) _
                    .AtIndex(1) _
                    .WithSignature(SignatureBuilder.SignatureFor("john.smith@email.com") _
                        .OnPage(0) _
                        .AtPosition(500, 100))) _
                .Build()

            Dim id = client.CreatePackage(package)

            client.SendPackage(id)

            Console.WriteLine("Package {0} was sent", id.Id)
        End Sub
    End Module
End Namespace